File: week_5/541_everse-string-ii/Solution.php
<?php


class Solution {

	/**
	 * @param String $s
	 * @param Integer $k
	 * @return String
	 */
	function reverseStr($s, $k) {
		for($start = 0; $start < strlen($s); $start += 2*$k){
			$i = $start;
			$j = min($start+$k-1, strlen($s)-1);
			while($i<$j){
				$tmp = $s[$i];
				$s[$i++] = $s[$j];
				$s[$j--] = $tmp;
			}
		}
		return $s;
	}
}

File: week_5/541_everse-string-ii/Solution.go
package main

import "fmt"

func reverseStr(s string, k int) string {
	var res = []byte(s)
	for start := 0; start < len(s); start += 2 * k {
		i := start
		var j int
		if (start + k - 1) < len(s)-1 {
			j = start + k - 1
		} else {
			j = len(s) - 1
		}
		for i < j {
			tmp := s[i]
			res[i] = s[j]
			res[j] = tmp
			i++
			j--
		}
	}
	return string(res)
}

func main() {
	fmt.Println(reverseStr("abcdefg", 2))
}
